src/engine/source/api/include/api/test/sessionManager.hpp

```diff
@@ -0,0 +1,86 @@
+#ifndef _SESSIONS_MANAGER_HPP
+#define _SESSIONS_MANAGER_HPP
+
+#include "iSessionManager.hpp"
+
+#include <shared_mutex>
+#include <unordered_map>
+
+namespace api::sessionManager
+{
+
+class SessionManager : public ISessionManager
+{
+private:
+    std::unordered_map<std::string, Session> m_activeSessions;
+    std::unordered_map<std::string, std::string> m_policyMap;
+    std::unordered_map<std::string, std::string> m_routeMap;
+
+    std::shared_mutex m_sessionMutex;
+
+public:
+    SessionManager() = default;
```

Comment:
improve-desing: These declarations are already made in the interface, please remove them